test/distributed/pipeline/sync/test_pipe.py

```diff
@@ -381,22 +381,124 @@ class NonTensorTuple(nn.Module):
         def forward(self, x):
             return (x, "hello")
 
+    class NonTensorArgs(nn.Module):
+        def forward(self, x: str, y: bool):
+            return x, y
+
     model = nn.Sequential(NonTensorTuple())
     model = Pipe(model)
     x = torch.rand(1)
 
-    # TypeError: CheckpointBackward.forward: expected Variable (got str) for return value 1
-    with pytest.raises(TypeError):
-        model(x)
-
-    # TypeError: expected Tensor to scatter, but got str
     with pytest.raises(TypeError):
         model((x, "hello"))
 
-    # TypeError: expected Tensor to scatter, but got str
     with pytest.raises(TypeError):
         model([x, "hello"])
 
+    model = nn.Sequential(NonTensorArgs())
+    model = Pipe(model)
+
+    with pytest.raises(TypeError):
+        model("hello", True)
```

Comment:
add a comment that the failure is because we  Need atleast one tensor? also can we enhance the error msg to indicate this?